lib/utmp.c

```diff
@@ -44,34 +44,30 @@
 
 #define UTX_LINESIZE  countof(memberof(struct utmpx, ut_line))
 
+// ttyname_ra - tty name re-entrant array
+#define ttyname_ra(fd, buf)  ttyname_r(fd, buf, countof(buf))
+
 
 /*
  * is_my_tty -- determine if "tty" is the same TTY stdin is using
  */
 static bool
 is_my_tty(const char tty[UTX_LINESIZE])
 {
-	char         full_tty[STRLEN("/dev/") + UTX_LINESIZE + 1];
-	/* tmptty shall be bigger than full_tty */
-	static char  tmptty[sizeof(full_tty) + 1];
+	char  full_tty[STRLEN("/dev/") + UTX_LINESIZE + 1];
+	char  my_tty[countof(full_tty)];
 
 	stpcpy(full_tty, "");
 	if (tty[0] != '/')
 		strcpy (full_tty, "/dev/");
 	strncat(full_tty, tty, UTX_LINESIZE);
```

Comment:
Hm, this isn't a result of your patch, and probably never really a problem,  but if tty happens to already start with /dev/, then we are restricting the copy to 5 fewer characters than if it does not.

Reply:
The source is not larger than `UTX_LINESIZE`, and the destination buffer we're using is `5 + UTX_LINESIZE + 1`, so we have plenty of size.